projects/fsastorefrontlib/src/cms-components/sync-pilot/sync-pilot-connection.component.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { UserService } from '@spartacus/core';
+import { CmsComponentData } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-fs-sync-pilot-connection-component',
+  templateUrl: './sync-pilot-connection.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SyncPilotConnectionComponent implements OnInit, OnDestroy {
+  constructor(
+    protected componentData: CmsComponentData<any>,
+    protected userService: UserService
+  ) {}
+
+  protected readonly CHANNEL_PARAM = '?c=';
+  protected readonly USER_PARAM = '&nick=';
+
+  component$: Observable<any>;
+
+  private subscription = new Subscription();
+
+  ngOnInit() {
+    this.component$ = this.componentData.data$;
+  }
+
+  establishConnection(targetUrl: string, channel: string, action: string) {
+    this.subscription.add(
+      this.userService
+        .get()
+        .pipe(
+          map(user => {
+            if (user?.uid && user?.name) {
+              const syncPilotUrl = targetUrl + action;
+              window.open(
```

Comment:
We shouldn't reference window object directly like this. This is because it's bypassing Angular's encapsulation and can lead to some security issues, so it's not considered as a good practice. Spartacus has windowRef service for this purpose which can be injected here instead.
Take a look at `tab-paragraph-container.component` as an example for this.

Reply:
Also we should check if in an addition to `_blank` there should be a `noopener` and `noreferrer"` passed as well, for security purposes. I remember that we had Sonar security issues because of this.

Reply:
Fixed. It opens a new tab using winRef service. _blank is not needed anymore. winRef demands 4 parameters btw, targetUrl, window name, additionalFeatures (for example "fullscreen=yes, toolbar=yes") & boolean value as indicator if new tab show be opened or stay at current.